crates/crates-io/lib.rs

```diff
@@ -25,6 +24,20 @@ pub struct Registry {
     auth_required: bool,
 }
 
+/// Represents a challenge of `Cargo` authentication scheme.
+#[non_exhaustive]
+#[derive(Debug, PartialEq, Eq)]
+pub enum Challenge {
+    /// With a `login_url=<URL>` parameter. Client is expected to login via the
+    /// provided URL.
+    Login(Url),
+    // With an one time token `token=<token>` parameter. Client is expected to
+    // respond this challenge with the given token. See [RFC 3231].
+    //
+    // [RFC 3231]: https://rust-lang.github.io/rfcs/3231-cargo-asymmetric-tokens.html#the-authentication-process
+    Token(String),
```

Comment:
parameter name bikeshedding.

* `token`
* `asymmetric_token`
* `one_time_token`

Reply:
or `one_time_challenge` or `nonce`

Reply:
`nonce` is the correct word I was forgetting!